modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,236 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+#include <algorithm>
+#include <iostream>
+#include <utility>
+#include <iterator>
+
+#include <opencv2/imgproc.hpp>
+
+namespace cv {
+namespace dnn {
+
+struct Model::Impl {
+    Size   size;
+    Scalar mean;
+    float  scale = 1.0;
+    bool   swapRB = false;
+    bool   crop = false;
+    Mat    blob;
+    std::vector<String> outNames;
+
+    void predict(Net& net, const Mat& frame, std::vector<Mat>& outs) {
+        if (size.width <= 0) size.width = frame.cols;
+        if (size.height <= 0) size.height = frame.rows;
```

Comment:
Let's for now ask user to provide input blob size in any case. So please throw an assertion if input size is now specified.